File: backend/src/main/java/com/kiwi/features/nodes/data/NodesDTO.java
package com.kiwi.features.nodes.data;

import lombok.*;

import java.util.List;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder @EqualsAndHashCode @ToString
public class NodesDTO {
    public Long id;
    public String status;
    public int icon;
    public int price;
    public float cordX;
    public float cordY;
    private String name;
    private String displayName;
    private List<Long> connectedNodeIds;
    private int mapId;
    private boolean isFirstNodeOfMap;
    private String onExecutionEvent;
    private int onExecutionEntityId;
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/nodes/data/NodesDTO.kt
package com.bellako.kiwi.features.nodes.data

data class NodesDTO(
    val id: Long,
    val icon: Int,
    val status: String,
    val price: Int,
    val cordX: Float,
    val cordY: Float,
    val name: String,
    val displayName: String,
    val connectedNodeIds: List<Long>,
    val mapId: Int,
    val onExecutionEvent: String,
    val onExecutionEntityId: Int,
    val transitionStyle: String = "VEIL",
)
